apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/class-wp-rest-wpcom-block-editor-nux-status-controller.php

```diff
@@ -79,10 +84,18 @@ public function get_nux_status() {
 		} else {
 			$nux_status = get_user_meta( get_current_user_id(), 'wpcom_block_editor_nux_status', true );
 		}
+
+		$show_editor_onboarding = $this->show_editor_onboarding( $nux_status );
+
 		return rest_ensure_response(
 			array(
-				'is_mobile_device' => $is_mobile_device,
-				'is_nux_enabled'   => $this->is_nux_enabled( $nux_status ),
+				'show_editor_onboarding'    => $show_editor_onboarding,
+				'editor_onboarding_variant' => $editor_onboarding_variant,
+
+				// These are legacy rest params that can be removed after
+				// we know the new JS files have been deployed.
+				'is_nux_enabled'            => $show_editor_onboarding,
```

Comment:
Wondering why you think this is a legacy reset param?  The `is_nux_enabled` is important for determining if the user is a new user.   It is a user preference that is set as 'enabled' for new users (as far as I understand).  So this isn't legacy, it will remain and is important so we can determine whether to automatically show the Welcome UI (whether Tour or NUX modal to the user).


(Side note: unfortunately this component is using the `is_nux_enabled` flag to also keep track of the open /closed status of the NUX (or [Tour](https://github.com/Automattic/wp-calypso/blob/578c58b8d687dbc28a59e4d06c2f875a655adfb3/apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/tour-launch.js#L78)), this is a code cleanup I'd like to do, use a flag purposely for open/close stus and keep the nux flags for the sole purpose of new-user-status')

Reply:
Ah!! I see what you did:
`	$show_editor_onboarding = $this->show_editor_onboarding( $nux_status );`

Should we carry that variable name change throughout the component (as now its a mixup of isNuxEnables and show_editor_onboarding:
https://github.com/Automattic/wp-calypso/blob/8e03ba31060da3b7fe1f6a7d23f4534bbc7bb9d7/apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/block-editor-nux.js#L43

https://github.com/Automattic/wp-calypso/blob/3b7743c86c77b3811e512fd351ad63e0b287ddde/apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/disable-core-nux.js#L21

https://github.com/Automattic/wp-calypso/blob/3b7743c86c77b3811e512fd351ad63e0b287ddde/apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/disable-core-nux.js#L25

And more like that in the NUX modal code and Tour code (e.g. dismiss events).

I think this would then satisfy my itch to have a client side flag that is about 'show' (and not about is_nux_enabled).   I'm just skimming all this so hopefully it all makes sense. :)

Reply:
Yes that's right, I wanted to rename that variable while I was at it, to make it more about "editor onboarding" and less about "nux". Sorry I hadn't updated the GH thread with any comments yet, they're all buried in the commit message.

I initially wanted to collapse it all into one variable: so something like `editor_onboarding_status` that would be either `false`, `'modal'` or `'tour'`. However I soon realised that the client needs to be able to open the onboarding itself even if `editor_onboarding_status` was `false` when the page loaded, and it needed to have an idea of which variant to load. So left it as two variables.

Yes I think it'd be good to carry the renaming through the rest of the store! I'll do that, but it'd be good to have a call to make sure I'm on the same page as you w.r.t. what sort of language we should be using. I'm thinking
- The umbrella feature is called "editor onboarding"
- The onboarding is either "shown" or "not shown"
- There are two "variants" of the onboarding

I didn't like the idea of using "is mobile" as part of the API. Not sure why, maybe just a taste thing? If felt like it's not high level what we're trying to communicate to the client. The server can make the decision around which variant to show and the client just does what it's told. It also opens up the idea of having `'tour_i1'`, `tour_i2`, etc. if we want to keep iterating.

Reply:
> all buried in the commit message.
A great commit message it is!!  Good example for the rest of us ! :)

> The umbrella feature is called "editor onboarding"
As we chatted about, indeed that is the umbrella feature, of which this Tour (or Guide as core calls it) may just be one part.   Additional help features may be included in Onboarding, besides just the Tour/Guide.

> I didn't like the idea of using "is mobile" as part of the API. Not sure why, maybe just a taste thing?
I first started by passing `show_welcome_tour` then switched to `is_mobile`, was still debating in my head if that was right, so good to hear your opinion and I fully support your changes! :)